src/components/CowCard/CowCard.test.js

```diff
@@ -342,4 +343,40 @@ describe('CowCard', () => {
       })
     })
   })
+
+  describe('custom naming', () => {
+    // NOTE: Validates the fix for:
+    // https://github.com/jeremyckahn/farmhand/issues/527
+    test('cows orignally owned by the player can be renamed', async () => {
+      const renderComponent = () => {
+        render(
+          <CowCard
+            {...{
+              ...baseProps,
+              cowInventory: [{ ...cow }],
+              isSelected: true,
+              // NOTE: This simulates how CowCard is integrated into the rest of
+              // the component tree. It also effectively reproduces the scenario
+              // that caused https://github.com/jeremyckahn/farmhand/issues/527
+              debounced: {
+                handleCowNameInputChange: () => renderComponent(),
+              },
+            }}
+          />
+        )
+      }
+
+      renderComponent()
+
+      const nameInput = screen.getByPlaceholderText('Name')
+
+      const customName = 'Custom'
+      userEvent.clear(nameInput)
+      userEvent.type(nameInput, customName)
+
+      await waitFor(() => {
```

Comment:
i think you should be able to await the userEvent.type calls and avoid waitFor entirely

Reply:
> i think you should be able to await the userEvent.type calls and avoid waitFor entirely

It turns out that `clear` and `type` only return Promises in `@testing-library/react` v14. Farmhand currently uses v13, and upgrading to v14 breaks a number of other tests so it's not worth making that change in a narrowly-scope bugfix PR such as this.

However, it appears that the `await waitFor(...)` isn't even necessary here (I think it was for an earlier iteration), so I've removed it in https://github.com/jeremyckahn/farmhand/commit/9601f41a3dae4b43975f742f1bde9ec726e4e72c. Thanks for bringing this up helping to simplify this part of the code!

Reply:
ah right, i forgot the switch to async was in later versions